### PR TITLE
syncer/checkpoint(dm): update the meaning of IsOlderThanTablePoint

### PR DESCRIPTION
**Pull request by lance6716**
### What problem does this PR solve?

Issue Number: ref #4287

### What is changed and how it works?

after https://github.com/pingcap/tiflow/pull/3891 , we can be confident to skip event <= table checkpoint when using position based replication

### Check List 

Tests 

 - pass original test

Code changes

 - Has exported function/method change

### Release note 

```release-note
`None`.
```

### CODE REVIEW

**Review by Ehco1996 — APPROVED** [review]
LGTM

**Review by GMHDBJD — APPROVED** [review]
LGTM

**Comment by lance6716** [comment]
/hold

found a bug in CI

**Comment by lance6716** [comment]
https://github.com/pingcap/tiflow/pull/4289/commits/dd9531495cc9aa9e8701e537438f6b3d7b679652

PTAL @Ehco1996 @GMHDBJD

**Comment by lance6716** [comment]
> can't understand this change, what if added2Queue = false but err2!=nil ?

it means `handleJob` meets error before send the job to DMLQueue or DDLQueue, which will not happen currently.

**Comment by Ehco1996** [comment]
> > can't understand this change, what if added2Queue = false but err2!=nil ?
> 
> it means `handleJob` meets error before send the job to DMLQueue or DDLQueue, which will not happen currently.

yes, but this could not hanppen in handleRowsEvent and how this change fix the `save checkpoint while job is not added to queue` 😵. could you please explain more

**Comment by lance6716** [comment]
I mean "save checkpoint while job is not added to queue" is a bug, the commit has fixed it.

File: dm/syncer/syncer.go [hunk 1]
```diff
@@ -1012,23 +1012,24 @@ func (s *Syncer) handleJob(job *job) (err error) {
 
 	if waitXIDStatus(s.waitXIDJob.Load()) == waitComplete && job.tp != flush {
```
Comment (Ehco1996) [inline dm/syncer/syncer.go:1013]: you mean in this line return nil-err and outer logic call s.checkShouldFlush() ? and this cause checkpoint have wrong pos ?

Comment (lance6716) [reply dm/syncer/syncer.go:1013]: yes.

**Pull request merged**